### ISSUE TITLE
What about validating user input and checking data for redundancy?

### ISSUE THREAD

**Issue by code2expert** [aspnet/Docs#11109]
Please enlighten further on how to validate user inputs before the form is submitted for example to check that the data in the title textbox is not already in the database to prevent redundancy.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 39b6d0f3-9bc7-d091-0d45-9fc1e2e7aac8
* Version Independent ID: 20e7eb53-215b-09ab-205d-d20177221278
* Content: [Add validation to an ASP.NET Core Razor Page](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/validation?view=aspnetcore-2.2)
* Content Source: [aspnetcore/tutorials/razor-pages/validation.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/tutorials/razor-pages/validation.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by code2expert** [comment]
Sorry, I found this feature in one of your earlier tutorials under "Remote Validation" with this link 
https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-2.2 

Thanks and keep up your good work.

**Comment by code2expert** [comment]
Oh oh..... Remote validation from the documentation seems to refer to controllers. 

Is it possible to do the validation in the model?

**Comment by code2expert** [comment]
How can I achieve more flexibility with validation especially with fields that can be initially be set to 0 but once a certain field has a number, some other field(s) become mandatory.

**Comment by Rick-Anderson** [comment]
Hello! Thanks for contacting us. This sounds like a general question about using ASP.NET Core. While we try to look at and respond to all issues, for questions like this we recommend posting to a community support group like [Stack Overflow](https://stackoverflow.com) with the `asp.net-core` tag.

**Comment by code2expert** [comment]
Hello Rick,

Thanks for this feedback, however I did not see this solution on stack overflow as suggested and I also noticed that a lot of people are coming up with the same question for Razor.

I also get the impression you believe stack overflow is extensive in it's solutions, while this is not a bad idea, I wouldn't go with you regarding upgraded technologies. I've practically been on stack overflow all day to no avail. 

I hope you find some time to have a look at the problem soon. Thanks once again.
